core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -872,6 +872,16 @@ public void writeList(List<? extends Writeable> list) throws IOException {
     }
 
     /**
+     * Writes a list of strings
+     */
+    public void writeStringList(List<String> list) throws IOException {
```

Comment:
Maybe this should just be `public void writeList(Writer<?> list) throws IOException {`?

If we did that we should more `Writer` out of `Writeable` and put it in `StreamOutput` I guess.... And we'd want a twin method in `StreamInput` and to move `Reader` there too I guess....

Maybe that should just wait until another PR.